torch/_inductor/compile_fx.py

```diff
@@ -424,6 +384,97 @@ def compiled_fn(new_inputs):
     return result
 
 
+def fx_codegen_and_compile(
```

Comment:
What's going on here?

Reply:
I split the `inner_compile_fx` function into 2: `fx_codegen_and_compile` and `inner_fx_compile`. The first function is the one that will be cached by the cache implemented in #103453, while the rest of `inner_fx_compile` contains all the cudagrahs code that can't be as easily cached due to the runtime requirement.

Reply:
Then I return the new object `CompiledFxGraph`, which is the one that can be cached. If you have a better name for the function splits, I'm happy to take suggestions.